### PR TITLE
fix: cross-covariance matrix in FullBilloirVertexFitter

### LINKED ISSUE #2231: Bug: Negative variances from AMVF
When running `CI/physmon/workflows/physmon_vertexing.py`, I encountered the following error (see screenshot): The variance of the fitted vertex from the AMVF is sometimes negative (the variance from the IVF is fine). 

When running `CI/physmon/workflows/physmon_vertexing.py` with the code from PR #2221, I find that the variance of the track parameters from the AMVF is also not always positive. 

image

### PR DESCRIPTION
**Pull request by felix-russo**
Fixes incorrect cross-covariance matrix 

Finally resolves #2231 

Note: Var(T) <= 0 is still encountered sometimes but this is because we don't always fit time. I will remove this warning in another PR

### CODE REVIEW

File: Core/include/Acts/Vertexing/FullBilloirVertexFitter.ipp [hunk 1]
```diff
@@ -261,9 +261,7 @@ Acts::FullBilloirVertexFitter<input_track_t, linearizer_t>::fit(
       transMat(5, 3) = 1.;
 
       // cov(V,P)
-      // TODO: This is incorrect (see Ref. (3)), but it will not be needed
-      // anyway once we replace fittedParams with fittedMomentum
-      ActsMatrix<4, 3> covVP = billoirTrack.B;
+      ActsMatrix<4, 3> covVP = -covV * billoirTrack.BCinv;
```
Comment (paulgessinger) [inline Core/include/Acts/Vertexing/FullBilloirVertexFitter.ipp:264]: Is there a corresponding equation for this in the literature somewhere? It's not obvious to me what's being calculated here, so maybe you could add an extra comment?

**Review by paulgessinger — COMMENTED** [review]
This seems to primarily change the pull of the fitted tracks (which is post-vertex-fit, right?

image

Does it makes sense that that's the only quantity that seems affected? To my eye the distribution becomes more gaussian shaped, which I suppose is good.

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [58b197539ba687b2c72afa9b15b56e5ae84f3d97]
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535629) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535628) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535627) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535626) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535625) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535624) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535623) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535622) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535621) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535620) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535619) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535618) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535617) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535616) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535615) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535614) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535613) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535612) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535611) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535610) |
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535609) |
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34535608) |

**Comment by paulgessinger** [comment]
Ok got it. So we update the references here and merge?

**Comment by felix-russo** [comment]
updated references @paulgessinger

**Comment by paulgessinger** [comment]
Hmm, Apogee reports Athena build fails, but I don't think it's because of this PR actually. I'll check manually.

**Review by paulgessinger — APPROVED** [review]

**Pull request merged**